src/stream/src/executor/backfill.rs

```diff
@@ -335,13 +355,18 @@ where
             )
             .await?;
 
+        let start = Instant::now();
         pin_mut!(iter);
 
         while let Some(data_chunk) = iter
             .collect_data_chunk(table.schema(), Some(CHUNK_SIZE))
             .instrument_await("backfill_snapshot_read")
             .await?
         {
+            if start.elapsed().as_millis() > flow_control_time_ms {
```

Comment:
I think you missed resetting start here? Otherwise we will sleep 1ms every chunk

Reply:
The `snapshot_read` will be called every epoch and dropped at the end of the barrier.